### DISCUSSION TITLE
Support non-ascii characters for index uid

### DISCUSSION THREAD

**Discussion by jet10000** [meilisearch/product#465] [Feedback & Feature Proposal]
meilisearch.errors.MeiliSearchApiError: MeiliSearchApiError. Error code: invalid_index_uid. Error message: `中文_123456` is not a valid index uid. Index uid can be an integer or a string containing only alphanumeric characters, hyphens (-) and underscores (_). Error documentation: https://docs.meilisearch.com/errors#invalid_index_uid Error type: invalid_request

**Comment by gmourier** [comment]
Hi @jet10000 👋

I guess it would be possible from a purely technical point of view. @Kerollmops do you confirm?

Now I have a usage question. Won't it be a problem to reach a resource in the API path if it contains non-ASCII chars?

Thanks!

**Reply by Kerollmops** [reply]
> I guess it would be possible from a purely technical point of view. @Kerollmops do you confirm?

Indeed, I don't see any technical issues. The engine is capable of handling any valid UTF-8 character.

> Now I have a usage question. Won't it be a problem to reach a resource in the API path if it contains non-ASCII chars?

This is the point where I find it a little bit more complex than before, it would require the users to correctly URL encode their index UIDs in the URL. This can be an issue but we can easily help users by identifying and explaining how to URL encode things.

As an example, the URL encoded version of your index UID `中文_123456` would become `%E4%B8%AD%E6%96%87_123456`.

In fact, if we support any valid UTF-8 string, it would open the engine to more people, in my point of view it is much better if we accept them. If we do we should also reconsider the restriction on the document IDs too. One argument that goes in the same direction is that we are making sure that we handle CJK languages when we tokenized the documents.
